apiserver/facades/client/machinemanager/machinemanager.go

```diff
@@ -358,35 +362,17 @@ func (mm *MachineManagerAPI) addOneMachine(ctx context.Context, p params.AddMach
 	return mm.st.AddMachineInsideNewMachine(template, template, p.ContainerType)
 }
 
-func (mm *MachineManagerAPI) saveMachineInfo(ctx context.Context, machineName, nonce string) error {
+func (mm *MachineManagerAPI) saveMachineInfo(ctx context.Context, nonce string) error {
 	// This is temporary - just insert the machine id all al the parent ones.
-	var errs []error
-	for machineName != "" {
-		var n *string
-		if nonce != "" {
-			n = &nonce
-		}
-		_, err := mm.machineService.CreateMachine(ctx, coremachine.Name(machineName), n)
-		// The machine might already exist e.g. if we are adding a subordinate
-		// unit to an already existing machine. In this case, just continue
-		// without error.
-		if err != nil && !errors.Is(err, machineerrors.MachineAlreadyExists) {
-			errs = append(errs, errors.Annotatef(err, "saving info for machine %q", machineName))
-		}
-		parent := names.NewMachineTag(machineName).Parent()
-		if parent == nil {
-			break
-		}
-		machineName = parent.Id()
-	}
-	if len(errs) == 0 {
-		return nil
+	var n *string
+	if nonce != "" {
+		n = &nonce
 	}
-	var errStr string
-	for _, e := range errs {
-		errStr += e.Error() + "\n"
+	createMachineArgs := machineservice.CreateMachineArgs{
+		Nonce: n,
```

Comment:
We're missing the directive here, otherwise we can't create child machines.

Reply:
Done, added.